### PR TITLE
0.14.1, pin ruamel.yaml, pip check, metadata

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes:
- alternative to/includes #152
- removes `name` jinja variable
- adds `pip check`
- adds pin for `ruamel.yaml`
- removes extra `build` environmment (doesn't seem to be needed for noarch packages)
- update docs URL to new location with https

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge/great-expectations ready for review!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -142,13 +138,31 @@ requirements:
     # - azure-identity >=1.0.0
     # - airflow >=1.9.0
     # - Click >=7.1.2
+    # - sqlalchemy >=1.3.16
+    # - snowflake-sqlalchemy >=1.2
+    # - ruamel_yaml <0.17.18,>=0.16
```
Comment (xhochy) [inline recipe/meta.yaml:143]: Ah, this matching failed!

Comment (bollwyvl) [reply recipe/meta.yaml:143]: oh is it automated or something?

Comment (xhochy) [reply recipe/meta.yaml:143]: Yes, the cf-scripts extract the requirements from the whl (if there is one) if you have set `bot.run_deps_from_wheel: true` in the conda_forge.yml.

**Pull request merged**